modules/ncc-spoke-ra/variables.tf

```diff
@@ -21,12 +21,17 @@ variable "data_transfer" {
 }
 
 variable "hub" {
-  description = "The name of the NCC hub to create or use."
+  description = "The name of the NCC hub to create or the id for an existing hub to use."
   type = object({
     create      = optional(bool, false)
     description = optional(string)
-    name        = string
+    id          = optional(string)
+    name        = optional(string)
   })
+  validation {
+    condition     = var.hub.create && var.hub.name != null || !var.hub.create && var.hub.id != null
```

Comment:
While it's a bit more lengthy I'd try to be more explicit.
btw I think hub name should be populated if hub create is true. Am I missing something?

For example:
`var.hub_create && var.hub.name == null || var.hub.create == false && var.hub.id == null`
```

Reply:
Made the condition more explicit replacing ! with == false, overall the condition LGTM (tested as well locally)